angular-chess/src/app/services/highlighting.service.ts

```diff
@@ -1,40 +1,41 @@
 import { Injectable } from '@angular/core';
 import { BehaviorSubject, map, Observable, tap } from 'rxjs';
-import { HighlightColor, Position, Square } from '../types/board.t';
+import { HighlightColor, SquareWithHighlight } from '../types/board.t';
+import { Square } from '../types/compressed.types.t';
 import LoggingUtils, { LogLevel } from '../utils/logging.utils';
-import PositionUtils from '../utils/position.utils';
+import SquareUtils from '../utils/square.utils';
 
 @Injectable({
   providedIn: 'root'
 })
 export class HighlightingService {
-  private squares$$: BehaviorSubject<Square[]> = new BehaviorSubject<Square[]>([]);
-  private squares$: Observable<Square[]> = this.squares$$.asObservable();
+  private squares$$: BehaviorSubject<SquareWithHighlight[]> = new BehaviorSubject<SquareWithHighlight[]>([]);
+  private squares$: Observable<SquareWithHighlight[]> = this.squares$$.asObservable();
 
-  public getSquares$(): Observable<Square[]> {
+  public getSquares$(): Observable<SquareWithHighlight[]> {
     return this.squares$.pipe(tap(data => LoggingUtils.log(LogLevel.INFO, () => `Square data: ${data}`)));
   }
 
-  public getSquare$(position: Position): Observable<Square | undefined> {
-    return this.squares$$.pipe(map((squares: Square[]) => squares.find(data => data.position.column === position.column && data.position.row === position.row)))
+  public getSquare$(position: Square): Observable<SquareWithHighlight | undefined> {
+    return this.squares$$.pipe(map((squares: SquareWithHighlight[]) => squares.find(data => SquareUtils.fileOf(data.position) === SquareUtils.fileOf(position) && SquareUtils.rankOf(data.position) === SquareUtils.rankOf(position))))
```

Comment:
Looks like a redundant function call. Please compare to chess-board.ts
![grafik](https://github.com/RubikNube/angular-chess/assets/97945202/0716d31a-de78-43a2-ba56-2e4c3c73cf39)

Perhaps you can create a readonly const compare function.